src/wasm/wasm-s-parser.cpp

```diff
@@ -2643,20 +2643,28 @@ Index SExpressionWasmBuilder::getStructIndex(Element& type, Element& field) {
   return atoi(field.c_str());
 }
 
+static void verifyPackedRead(const Field& field,
+                             SExpressionWasmBuilder::Packed packed,
+                             Element& s,
+                             std::string what) {
```

Comment:
The `what` could probably just be "field" in both cases, since the line and column will identify the precise culprit.

Reply:
Hmm, but Array has `element` which I've assumed was intentional?

Regardless, for the error message I think we want to say if it's a struct or an array, so the text should differ I think.